### PR TITLE
Enable video at low res by default when adaptive stream is enabled.

### PR DESCRIPTION
**Pull request by davidzhao**
When AdaptiveStream is enabled, default the subscriber to LOW quality stream we would want LOW instead of OFF for a couple of reasons
1. when a subscriber unsubscribes from a track, we would forget their previously defined settings depending on client implementation, subscription on/off is kept separately from adaptive stream So when there are no changes to desired resolution, but the user re-subscribes, we may leave stream at OFF
2. when interacting with dynacast *and* adaptive stream. If the publisher was not publishing at the time of subscription, we might not be able to trigger adaptive stream updates on the client side (since there aren't any video frames coming through). this will leave the stream "stuck" on off, without a trigger to re-enable it

### CODE REVIEW

**Review by paulwe — APPROVED** [review]
lgtm

**Review by cnderrauber — APPROVED** [review]

File: pkg/rtc/subscribedtrack.go [hunk 1]
```diff
@@ -82,11 +82,30 @@ func (t *SubscribedTrack) Bound() {
...
+		if t.params.AdaptiveStream {
+			desiredLayer = buffer.VideoQualityToSpatialLayer(livekit.VideoQuality_LOW, t.params.MediaTrack.ToProto())
+		} else {
+			desiredLayer = buffer.VideoQualityToSpatialLayer(livekit.VideoQuality_HIGH, t.params.MediaTrack.ToProto())
+		}
+		settings := t.settings.Load()
+		if settings != nil {
+			desiredLayer = t.spatialLayerFromSettings(settings)
+		}
+		t.DownTrack().SetMaxSpatialLayer(desiredLayer)
```
Comment (boks1971) [inline pkg/rtc/subscribedtrack.go:106]: Great catch on the second case.

Wondering if there is a change in first case. Maybe, subscription manager is losing the settings? The settings used to be cached in participant (it was actually done for migration IIRC, but was later useful in other cases too) and applied when a subscribed track was added. Something equivalent to [this](https://github.com/livekit/livekit/blob/db40272657b5205afb122d49e20eb6b4cc690dbe/pkg/rtc/subscriptionmanager.go#L643) in subscription manager. But, I have to check if subscription manager can persist this across unsubscription/subscription. Looks like it is in the `trackSubscription` struct which might be wiped out on unsubscription. Have to think about side effects of it getting wiped (as I think we should probably preserve that for migration cases???). Maybe required to hold the subscriber settings at subscription manager level and restore when a subscription is desired.

**Review by boks1971 — APPROVED** [review]

File: pkg/rtc/subscribedtrack.go (hunk 1 above)
Comment (davidzhao) [reply pkg/rtc/subscribedtrack.go:106]: It is a change in behavior compared to before. But I don't believe it should be persisting subscription settings across an unsubscribe. It was storing it on the participant before, which persisted it as a side effect IMO.

An unsubscribe means the subscriber is uninterested in the track. new subscription requests should come with a new subscription settings request.

Comment (boks1971) [reply pkg/rtc/subscribedtrack.go:106]: Makes sense. I guess in migration case, there should be no real "unsubscribe" and hence the settings should not be lost as it should only be switching the provider if necessary.

Comment (davidzhao) [reply pkg/rtc/subscribedtrack.go:106]: yeah, this is only including explicit user unsubscribes. I think on migration, clients would resend up existing subscription settings. So I think we are ok there on the new instance.

Comment (boks1971) [reply pkg/rtc/subscribedtrack.go:106]: Actually, not talking about the new instance case. Am referring to transparent subscription move. Let us say `pA` and `pB` are both on `nodeA`. `pB` is subscribed to `pA`. `pA` migrates to `nodeB`. That is transparent to `pB`. So, `pB`'s subscription of `pA` should move from local track -> remote track (the provider changes). Vice-versa if `pA` migrates into `nodeA`. Down track was being deleted from one provider and re-created/re-attached to another provider. The setting need to be re-applied on the new down track. That's the case I was referring to. That is the original reason for caching those settings in the older implementation.

Comment (davidzhao) [reply pkg/rtc/subscribedtrack.go:106]: yup, understood. When the publisher migrates, it does not trigger an explicit unsubscribe from the subscriber's perspective. So in that case the subscription struct will be left alone. We will only clear it out on an *explicit* unsubscribe request from the user. (or if the publisher has left for awhile and does not come back)

Comment (boks1971) [reply pkg/rtc/subscribedtrack.go:106]: Awesome! Thank you for the explanation.

**Pull request merged**